typing/btype.ml

```diff
@@ -84,14 +84,12 @@ type changes =
   | Unchanged
   | Invalid
 
-let trail = s_table Weak.create 1
+let trail = s_ref (ref Unchanged)
```

Comment:
I think that `s_table ref Unchanged` and `s_ref (ref Unchanged)` don't have the same semantics, and you want the `s_table` version here (the problem is that `s_ref`, when asked to re-initialize the store, will use the initial value, which would create buggy mutable aliasing here). cc @trefis for confirmation.

If my understanding is correct, then the documentation should probably emphasize that `s_ref` is to be used with pure values only -- or possibly consider removing it and using `s_table Fun.id v` in the pure case.

Reply:
Correct!

Reply:
I didn't know that. Now fixed.